```

gdal2tiles.py -k --s_srs EPSG:3857 -t "ten-forward" -g "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ" public/warps/ten-forward/ten-forward-geo.tif public/tms/ten-forward/
ERROR 6: EPSG PCS/GCS code 900913 not found in EPSG support files.  Is this a valid
EPSG coordinate system?
ERROR 6: No translation for an empty SRS to PROJ.4 format is known.
ERROR 6: No translation for an empty SRS to PROJ.4 format is known.
Traceback (most recent call last):
  File "/usr/bin/gdal2tiles.py", line 2262, in <module>
    gdal2tiles.process()
  File "/usr/bin/gdal2tiles.py", line 475, in process
    self.generate_metadata()
  File "/usr/bin/gdal2tiles.py", line 1123, in generate_metadata
    f.write( self.generate_tilemapresource())
  File "/usr/bin/gdal2tiles.py", line 1519, in generate_tilemapresource
    for z in range(self.tminz, self.tmaxz+1):
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
F
Finished in 14.985569s, 0.0667 runs/s, 0.4671 assertions/s.
  1) Failure:
ExporterTest#test_all_functions [test/exporter_test.rb:92]:
Expected false to be truthy.
1 runs, 7 assertions, 1 failures, 0 errors, 0 skips
The command "ruby test/exporter_test.rb" exited with 1.
```

--- a/lib/mapknitterExporter.rb
+++ b/lib/mapknitterExporter.rb
@@ -290,7 +290,8 @@ class MapKnitterExporter
   def self.generate_tiles(key, slug, root)
     key = "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ" if key == "" # ugh, let's clean this up!
     key = key || "AIzaSyAOLUQngEmJv0_zcG1xkGq-CXIPpLQY8iQ"
-    gdal2tiles = 'gdal2tiles.py -k --s_srs EPSG:3857 -t "'+slug+'" -g "'+key+'" '+'public/warps/'+slug+'/'+slug+'-geo.tif '+'public/tms/'+slug+"/"
+    zooms = zooms || "8-10"
+    gdal2tiles = "gdal2tiles.py -k -z #{zooms} --s_srs EPSG:3857 -t #{slug} -g #{key} public/warps/#{slug}/#{slug}-geo.tif public/tms/#{slug}/"
     puts gdal2tiles
     system(self.ulimit+gdal2tiles)
   end
@@ -299,13 +300,13 @@ class MapKnitterExporter
   def self.zip_tiles(slug)
     rmzip = 'cd public/tms/ && rm '+slug+'.zip && cd ../../'
     system(rmzip)
-    zip = 'cd public/tms/ && ' + self.ulimit + 'zip -rq '+slug+'.zip '+slug+'/ && cd ../../'
+    zip = "cd public/tms/ && #{self.ulimit}zip -rq #{slug}.zip #{slug}/ && cd ../../"
     system(zip)
   end
 
   # generates a tileset at public/tms/<slug>/
   def self.generate_jpg(slug, root)
-    imageMagick = 'convert -background white -flatten '+root+'/public/warps/'+slug+'/'+slug+'-geo.tif '+root+'/public/warps/'+slug+'/'+slug+'.jpg'
+    imageMagick = "convert -background white -flatten #{root}/public/warps/#{slug}/#{slug}-geo.tif #{root}/public/warps/#{slug}/#{slug}.jpg"
     system(self.ulimit+imageMagick)
   end
 

